src/ExternalModule.ts

```diff
@@ -38,6 +38,8 @@ export default class ExternalModule {
 			get dynamicImporters() {
 				return dynamicImporters.sort();
 			},
+			exportedBindings: {},
+			exports: [],
```

Comment:
I wonder if we could rather do this lazily with getters rather than imperatively. The thinking here is that the current solution requires us to remember that whenever we do something with `exports`, `reexportDescriptions` or `exportAllModules`, we need to remember to also update the fields on info. Also, we do not expect this information to be needed regularly.
Instead if this information is derived just in time from the `exports`, `reexportDescriptions` and `exportAllModules` we would have a single source of truth of sorts. And the information would also be protected from being overwritten.

Reply:
Yeah, I agree with you, I will do it.

Reply:
Currently, `ExternalModule` class does not contain `exports`,`reexportDescriptions`or`exportAllModules`. So the value of `exportedBindings` or `exports` could always be the initial value(`{}`or`[]`)?

Reply:
Hmm, either that or we explicitly set those values to `null` to signify that we do not know anything about external modules.
Alternatively, we can use `ExternalModule.exportedVariables.values()` to get a list of which values are actually imported from an external module and put that list into `exports`. `exportedBindings`, on the other hand, would still not really make sense because we still do not know where those exports came from. What do you think=

Reply:
Hmm, Currently I tend to set those values to `null` under the related issue #4721. the `moduleParsed` hook can not access `ExternalModule`.

Reply:
Yes and no. Any hook can use `this.getModuleInfo` to query information about any module including external ones. But `null` would be fine with me.

Reply:
Oh, I get it. By the way, the test coverage is lower than before, we might need to add tests for this code https://github.com/rollup/rollup/pull/4731/files#annotation_6576565827
Which file location is suitable for writing this tests?

Reply:
For plugin interface tests, I would usually write a `function` test. This is also the default category I recommend for tests unless you want to explicitly snapshot the generated code (e.g. to test tree-shaking stuff).
`function` tests bundle the code and then run it once. In your case, the running is not really important, so just put some dummy code into the files. Do not put console.log statements in the code as they will mess up tests output. `assert.ok(true)` is a good statement that is valid (Node's "assert" is injected as a global variable into `function` tests) and will not be removed by tree-shaking (which avoids an "empty bundle" warning).

Reply:
Ok, i get it, thanks for your guide.